lib/SILOptimizer/Transforms/SemanticARCOpts.cpp

```diff
@@ -1659,7 +1677,8 @@ class StorageGuaranteesLoadVisitor
 
     // If we have a modify, check if our value is /ever/ written to. If it is
     // never actually written to, then we convert to a load_borrow.
-    return answer(ARCOpt.isAddressWrittenToDefUseAnalysis(access));
+    auto result = ARCOpt.isAddressWrittenToDefUseAnalysis.get(access);
+    return answer(!result.hasValue() || result.getValue().size());
```

Comment:
I can't make sense of this code since it's written in terms of the multi-map interface. What does it mean for the analysis to have or not have a value, and how is that different from it being empty or having multiple elements?